modules/drivers/bigquery/src/metabase/driver/bigquery.clj

```diff
@@ -144,10 +144,18 @@
 
 (def ^:private ^:const ^Integer query-timeout-seconds 60)
 
+(def ^:private ^:dynamic ^Long max-results
+  "Maximum number of rows to return per request in a query."
+  20000)
```

Comment:
we have an `absolute-max-results` constant defined somewhere, does it make sense to use that instead of having a different max for BigQuery?

Also, I think 20k is a little low... some people might want to download a huge dataset

Reply:
this is per page, not total - let me rename it for clarity to `max-results-per-page`

Reply:
Is the new pagination getting past the BQ API limitation of 10MB? Or would it be better to apply a smaller page max?
https://cloud.google.com/bigquery/quotas#jobsquery_requests

Reply:
So, `max-results-per-page` doesn't overrule the 10MB limit. Pages can be a maximum of 10MB, or `max-results-per-page`, whichever threshold is met first.

On the HN dataset that Damon has, I would get approximately 22,000 rows per page.

With this change (pagination), I managed to download a CSV approximately 600 mb in size before the dev-environment timeout of 30 seconds hit.

In BigQuery, the "best" way to do a query is to submit the query as a job, and then subsequently call the GetQueryResults API repeatedly to paginate through the results with that job ID and project ID.

Reply:
Okay, OP on #6960 was only getting 1,759 rows, so that was just why I wanted to ask.